core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMaterializedViewDefinition.java

```diff
@@ -13,8 +13,6 @@
  */
```

Comment:
Extract this into three commits:
1. Commit that decouples SPI MV from Iceberg MV
2. Commit that simplifies SPI MV (including non-optional owner)
3. Commit that removes Json annotations in SPI MV. This could be separate PR potentially to let other maintainers to review as it changes how we should model other POJO SPI classes.